### ISSUE TITLE
Integral not evaluated using integration by parts

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#15431]
Sympy is unable to evalue this integral:
```julia
In [26]: integrate(x*exp(x)*log(x), x)
Out[26]: 
⌠               
⎮    x          
⎮ x⋅ℯ ⋅log(x) dx
⌡            
```

It can be evaluated using integration by parts. Taking `dv = x*exp(x)` and `u = log(x)` we get:
```julia
In [31]: u = log(x)

In [32]: dv = x*exp(x)

In [33]: integrate(u * dv, x)
Out[33]: 
⌠               
⎮    x          
⎮ x⋅ℯ ⋅log(x) dx
⌡               

In [34]: u * v - integrate(integrate(dv, x) * diff(u, x), x)
Out[34]: 
         x           x        
(x - 1)⋅ℯ ⋅log(x) - ℯ  + Ei(x)

In [35]: diff(_, x)
Out[35]: 
                                              x    x
         x           x           x   (x - 1)⋅ℯ    ℯ 
(x - 1)⋅ℯ ⋅log(x) + ℯ ⋅log(x) - ℯ  + ────────── + ──
                                         x        x 

In [36]: simplify(_)
Out[36]: 
   x       
x⋅ℯ ⋅log(x)
```

**Comment by avishrivastava11** [comment]
@oscarbenjamin I don't understand why there was a need for these steps :
```
In [35]: diff(_, x)
Out[35]: 
                                              x    x
         x           x           x   (x - 1)⋅ℯ    ℯ 
(x - 1)⋅ℯ ⋅log(x) + ℯ ⋅log(x) - ℯ  + ────────── + ──
                                         x        x 

In [36]: simplify(_)
Out[36]: 
   x       
x⋅ℯ ⋅log(x)
```
The evaluated integral is the output in `Out[34]`. So why did you feel the need to differentiate again?

**Comment by avishrivastava11** [comment]
@oscarbenjamin  Also, good job finding this out. I think this will be an important functionality for sympy (integral using by parts)

P.S. I've started working on this.

**Comment by oscarbenjamin** [comment]
> The evaluated integral is the output in `Out[34]`. So why did you feel the need to differentiate again?

I just did that to demonstrate that the result was an antiderivative for the original integral.

**Comment by asmeurer** [comment]
Only manualintegrate uses integration by parts, but I don't think it handles Ei yet. It also uses the LIATE rule so that might need to be modified.

**Comment by oscarbenjamin** [comment]
This issue doesn't seem to have been solved by #15619:
```julia
In [1]: integrate(x*exp(x)*log(x), x)                                                                                                          
Out[1]: 
⌠               
⎮    x          
⎮ x⋅ℯ ⋅log(x) dx
⌡               

In [2]: integrate(x*exp(x)*log(x), x, manual=True)                                                                                             
Out[2]: 
⌠               
⎮    x          
⎮ x⋅ℯ ⋅log(x) dx
⌡  
```
